equinox/nn/_composed.py

```diff
@@ -136,32 +139,58 @@ class Sequential(Module):
         Activation functions can be added by wrapping them in [`equinox.nn.Lambda`][].
     """
 
-    layers: tuple[_SequentialLayer, ...]
+    layers: tuple
 
-    def __init__(self, layers: Sequence[_SequentialLayer]):
+    def __init__(self, layers: Sequence[Module]):
         self.layers = tuple(layers)
 
+    @overload
     def __call__(self, x: Array, *, key: Optional[PRNGKeyArray] = None) -> Array:
+        ...
+
+    @overload
+    def __call__(
+        self, x: Array, state: State, *, key: Optional[PRNGKeyArray] = None
+    ) -> tuple[Array, State]:
+        ...
+
+    def __call__(
+        self,
+        x: Array,
+        state: State = sentinel,
+        *,
+        key: Optional[PRNGKeyArray] = None,
+    ) -> Union[Array, tuple[Array, State]]:
         """**Arguments:**
 
-        - `x`: Argument passed to the first member of the sequence.
-        - `key`: A `jax.random.PRNGKey`, which will be split and passed to every layer
-            to provide any desired randomness. (Optional. Keyword only argument.)
+        - `x`: passed to the first member of the sequence.
+        - `state`: If provided, then it is passed on to any layer which implements
+            `equinox._batch_norm._StatefulLayer`.
+        - `key`: Ignored; provided for compatibility with the rest of the Equinox API.
+            (Keyword only argument.)
 
         **Returns:**
-
         The output of the last member of the sequence.
+
+        If `state` is passed, then a 2-tuple of `(output, state)` is returned.
+        If `state` is not passed, then just the output is returned.
         """
 
         if key is None:
             keys = [None] * len(self.layers)
         else:
             keys = jrandom.split(key, len(self.layers))
         for layer, key in zip(self.layers, keys):
-            x = layer(x, key=key)
-        return x
+            if isinstance(layer, StatefulLayer):
+                x, state = layer(x, state=state, key=key)
+            else:
+                x = layer(x, key=key)
+        if state is sentinel:
+            return x
+        else:
+            return x, state
 
-    def __getitem__(self, i: Union[int, slice]) -> _SequentialLayer:
+    def __getitem__(self, i: Union[int, slice]) -> Module:
```

Comment:
Likewise the return annotation here should actually be `Callable`, not `Module`, I've realised.